src/cmds/pbsdsh.c

```diff
@@ -95,6 +107,96 @@ bailout(int sig)
  * @return - Void
  *
  */
+
+
+int
+host_match(char *line, char *host)
+{
+	if (NULL == host ) return 0;
+	int len = strlen(line);
+	static char domain[PBS_MAXHOSTNAME + 1];
+	char fullhost[PBS_MAXHOSTNAME + 1];
+	static struct in_addr addr;
+	static int addrvalid = -1;
+
+	if (line[len - 1] == '\n')
+		line[len - 1] = '\0';
+
+	if (strcmp(line, host) == 0)
+		return 1;
+
+	if (addrvalid == -1) {
+		addr.s_addr = inet_addr(host);
+		addrvalid = (addr.s_addr == INADDR_NONE) ? 0 : 1;
+	}
+
+	if (addrvalid) { /* compare IP addresses */
+		struct hostent *hp = gethostbyname(line);
+		int i;
+
+		if (hp == NULL)
+			return 0;
+
+		for (i = 0; hp->h_addr_list[i]; i++) {
+			if (memcmp(&addr, hp->h_addr_list[i],
+				   hp->h_length) == 0)
+				return 1;
+		}
+		return 0;
+	}
+
+	if (domain[0] == '\0') {
+		if (getdomainname(domain, (sizeof(domain) - 1)) == -1) {
+			perror("getdomainname");
+			exit(255);
+		}
+		if (domain[0] == '\0') {
+			int i;
+			char *dot;
+
+			if (gethostname(domain, (sizeof(domain) - 1)) == -1) {
+				perror("getdomainname");
+				exit(255);
```

Comment:
Should this be gethostname instead of getdomainname in the perror()?

Reply:
i believe it is a typo, could fix it, note again its also in pbs_tmrsh.